### DISCUSSION TITLE
LPD8806 RGB + CCT possibility

### DISCUSSION THREAD

**Discussion by lolimpol** [Makuna/NeoPixelBus#762] [Q&A] [answered]
Hi! I'm wondering if the LPD8806 can be used as an RGB + CCT chip like the FW1906.
It should be a superior chip, considering it can do 10 bit dimming, and has a 4000hz PWM frequency instead of 2600hz

Is it possible to do this? I want to use it with WLED.
And what about something like a UCS9812, that has 12 channels and 16bit dimming, that would be even better in my understanding.

**Comment by Makuna** [comment, marked as answer]
There is already support for [LPD8806](https://github.com/Makuna/NeoPixelBus/wiki/DotStar-Features#lpd8806brgfeature), but only for RGB.  Did you try it (even pixels rgb, odd pixels www for a test)?  It's a Two wire chip (data and clock).  You are asking about extending it to support Rgbwww then?  10 bit dimming on the LPD8806, that's not correct.  It only uses 7 bits, so not even the normal 8 bits.  They does NOT take a global dimming value like some other two wire LEDs.

Adding Rgbwxx.Rgbwwx/Rgbwww support is pretty trivial.  But what do you mean by + CCT?  This is a marketing term for app software that premixes colors into the white.  Not something this library does, it leaves that up to you when you specify the color with the RGB and (1-3) WWW channels.

Find the spec sheets for UCS9812, link them, and I can provide a better answer to how hard it is to support.

**Reply by lolimpol** [reply]
RGB CCT (CCT = Correlated Color Temperature) is just another way of saying rgbwww or RGBTW. Anyway, I didn't know that about the LPD8806. Seems the FW1906 is easier to use and provides the same result.

The UCS9812 is linked on a couple LED websites like this one: [https://www.advateklighting.com/pixel-protocols/ucs9812](https://www.advateklighting.com/pixel-protocols/ucs9812) and [http://www.szledcolor.com/news/view/id/460](http://www.szledcolor.com/news/view/id/460)

However I can't seem to find the bare chip for sale anywhere, and neither the datasheet. Seems thats not an option either.

Thanks for the help!

**Reply by Makuna** [reply]
Yeah, the 7bits per channel of the LPD8806 pretty much means it's a dinosaur.  I wonder if they have revised it with one that is truly 8bits per channel.

The UCS9812 looks very interesting.  12 channels per chip with 16bit resolution.

Just noting what I gleamed from those two links.
Pulse coding:
T0H high time 0.3μs 
T1H high time 0.6μs 
T0L low time 0.6μs 
T1L low time 0.3μs 
Treset low time 500us 

Data coding:
Data, 20 bits per channel
b19 - b 4      16 bit grey scale
b3 - b0         current setting, the values range from 2.4mA-20.4mA, in steps of 1.2mA

This seems to allow current settings per channel per chip; which is a little unusual and really not that useful for most designs.  My API currently only allows setting these per channel values across all the chips the same.  Following my current models, of all chips the same, this isn't that hard to support.  Let me know if you ever get any.  Setting current per chip is significantly more as I just don't expose any concept like it.

**Reply by lolimpol** [reply]
I've found an alternative in the MY9221, readily available, 16 bit 12 channels, 1000hz PWM. Basically the exact same as the USC9812 except I can find documentation about it. I'll start a new topic for it!
